File: src/main/java/com/anod/appwatcher/installed/ImportDataProvider.java
package com.anod.appwatcher.installed;

import android.content.Context;
import android.support.v4.util.SimpleArrayMap;

import com.anod.appwatcher.adapters.AppViewHolderDataProvider;
import com.anod.appwatcher.utils.InstalledAppsProvider;

class ImportDataProvider extends AppViewHolderDataProvider {

    static final int STATUS_DEFAULT = 0;
    static final int STATUS_IMPORTING = 1;
    static final int STATUS_DONE = 2;
    static final int STATUS_ERROR = 3;

    private SimpleArrayMap<String, Boolean> mSelectedPackages = new SimpleArrayMap<>();
    private boolean mDefaultSelected;
    private SimpleArrayMap<String, Integer> mProcessingPackages = new SimpleArrayMap<>();
    private boolean mImportStarted;

    ImportDataProvider(Context context, InstalledAppsProvider installedAppsProvider) {
        super(context, installedAppsProvider);
    }

    void selectAllPackages(boolean select) {
        mSelectedPackages.clear();
        mDefaultSelected = select;
    }

    void selectPackage(String packageName, boolean select) {
        mSelectedPackages.put(packageName, select);
    }

    boolean isPackageSelected(String packageName) {
        if (mSelectedPackages.containsKey(packageName)) {
            return mSelectedPackages.get(packageName);
        }
        return mDefaultSelected;
    }

    int getPackageStatus(String packageName) {
        if (mProcessingPackages.containsKey(packageName)) {
            return mProcessingPackages.get(packageName);
        }
        return STATUS_DEFAULT;
    }

    void setPackageStatus(String packageName, int status) {
        mProcessingPackages.put(packageName, status);
    }

    void setImportStarted(boolean started) {
        this.mImportStarted = started;
    }

    boolean isImportStarted() {
        return mImportStarted;
    }
}

File: src/main/java/com/anod/appwatcher/installed/ImportDataProvider.kt
package com.anod.appwatcher.installed

import android.content.Context
import android.support.v4.util.SimpleArrayMap
import info.anodsplace.framework.content.InstalledApps
import com.anod.appwatcher.watchlist.AppViewHolderDataProvider

internal class ImportDataProvider(context: Context, installedApps: InstalledApps) : AppViewHolderDataProvider(context, installedApps) {

    private val selectedPackages = SimpleArrayMap<String, Boolean>()
    private var defaultSelected: Boolean = false
    private val processingPackages = SimpleArrayMap<String, Int>()
    var isImportStarted: Boolean = false

    fun selectAllPackages(select: Boolean) {
        selectedPackages.clear()
        defaultSelected = select
    }

    fun selectPackage(packageName: String, select: Boolean) {
        selectedPackages.put(packageName, select)
    }

    fun isPackageSelected(packageName: String): Boolean {
        if (selectedPackages.containsKey(packageName)) {
            return selectedPackages.get(packageName)
        }
        return defaultSelected
    }

    fun getPackageStatus(packageName: String): Int {
        if (processingPackages.containsKey(packageName)) {
            return processingPackages.get(packageName)
        }
        return STATUS_DEFAULT
    }

    fun setPackageStatus(packageName: String, status: Int) {
        processingPackages.put(packageName, status)
    }

    companion object {
        const val STATUS_DEFAULT = 0
        const val STATUS_IMPORTING = 1
        const val STATUS_DONE = 2
        const val STATUS_ERROR = 3
    }
}